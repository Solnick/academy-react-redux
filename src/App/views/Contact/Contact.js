import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ContactForm from './ContactForm/ContactForm';
import MessageSentBanner from './MessageSentBanner/MessageSentBanner';
import View from '../../components/View/View';
import Tile from '../../components/Tile/Tile';

const Contact = ({
  isAlreadySent,
}) => (
  <View>
    <Tile center>
      {
        isAlreadySent
          ? <MessageSentBanner />
          : <ContactForm />
      }
    </Tile>
  </View>
);

Contact.propTypes = {
  isAlreadySent: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  isAlreadySent: state.contact.isAlreadySent,
});

export default connect(mapStateToProps)(Contact);
