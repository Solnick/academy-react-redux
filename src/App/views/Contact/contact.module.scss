@import '../../style/breakpoints';

.contactFormWrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 230px;
  margin: 50px 45px;
  @include on("medium") {
    width: 250px;
    margin: 50px 120px;
  }
  .sendButton {
    align-self: flex-end;
    margin-top: 30px;
    width: 90px;
  }
}