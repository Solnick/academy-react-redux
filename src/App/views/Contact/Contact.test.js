import React from 'react';
import { shallow } from 'enzyme';
import Contact from './Contact';

describe('Contact component', () => {
  it('should render correctly', () => {
    const contactInstance = shallow(<Contact />);
    expect(contactInstance).toMatchSnapshot();
  });
});
