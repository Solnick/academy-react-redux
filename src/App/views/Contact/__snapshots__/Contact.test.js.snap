// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contact />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Tile
        center={true}
      >
        <ValidatorForm
          debounceTime={0}
          onError={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="contactFormWrapper"
          >
            <TextValidator
              errorMessages="error"
              label="Name"
              name="name"
              onChange={[Function]}
              required={true}
              validatorListener={[Function]}
              validators={Array []}
              value=""
              variant="standard"
            />
            <TextValidator
              errorMessages={
                Array [
                  "Email is not valid",
                ]
              }
              label="Email"
              name="email"
              onChange={[Function]}
              required={true}
              validatorListener={[Function]}
              validators={
                Array [
                  "isEmail",
                ]
              }
              value=""
              variant="standard"
            />
            <TextValidator
              className="textArea"
              errorMessages="error"
              label="Message"
              multiline={true}
              name="message"
              onChange={[Function]}
              rows={6}
              validatorListener={[Function]}
              validators={Array []}
              value=""
              variant="standard"
            />
            <WithStyles(Button)
              className="sendButton"
              color="primary"
              type="submit"
              variant="contained"
            >
              Send
            </WithStyles(Button)>
          </div>
        </ValidatorForm>
      </Tile>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "center": true,
        "children": <ValidatorForm
          debounceTime={0}
          onError={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="contactFormWrapper"
          >
            <TextValidator
              errorMessages="error"
              label="Name"
              name="name"
              onChange={[Function]}
              required={true}
              validatorListener={[Function]}
              validators={Array []}
              value=""
              variant="standard"
            />
            <TextValidator
              errorMessages={
                Array [
                  "Email is not valid",
                ]
              }
              label="Email"
              name="email"
              onChange={[Function]}
              required={true}
              validatorListener={[Function]}
              validators={
                Array [
                  "isEmail",
                ]
              }
              value=""
              variant="standard"
            />
            <TextValidator
              className="textArea"
              errorMessages="error"
              label="Message"
              multiline={true}
              name="message"
              onChange={[Function]}
              rows={6}
              validatorListener={[Function]}
              validators={Array []}
              value=""
              variant="standard"
            />
            <WithStyles(Button)
              className="sendButton"
              color="primary"
              type="submit"
              variant="contained"
            >
              Send
            </WithStyles(Button)>
          </div>
        </ValidatorForm>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="contactFormWrapper"
          >
            <TextValidator
              errorMessages="error"
              label="Name"
              name="name"
              onChange={[Function]}
              required={true}
              validatorListener={[Function]}
              validators={Array []}
              value=""
              variant="standard"
            />
            <TextValidator
              errorMessages={
                Array [
                  "Email is not valid",
                ]
              }
              label="Email"
              name="email"
              onChange={[Function]}
              required={true}
              validatorListener={[Function]}
              validators={
                Array [
                  "isEmail",
                ]
              }
              value=""
              variant="standard"
            />
            <TextValidator
              className="textArea"
              errorMessages="error"
              label="Message"
              multiline={true}
              name="message"
              onChange={[Function]}
              rows={6}
              validatorListener={[Function]}
              validators={Array []}
              value=""
              variant="standard"
            />
            <WithStyles(Button)
              className="sendButton"
              color="primary"
              type="submit"
              variant="contained"
            >
              Send
            </WithStyles(Button)>
          </div>,
          "debounceTime": 0,
          "onError": [Function],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextValidator
                errorMessages="error"
                label="Name"
                name="name"
                onChange={[Function]}
                required={true}
                validatorListener={[Function]}
                validators={Array []}
                value=""
                variant="standard"
              />,
              <TextValidator
                errorMessages={
                  Array [
                    "Email is not valid",
                  ]
                }
                label="Email"
                name="email"
                onChange={[Function]}
                required={true}
                validatorListener={[Function]}
                validators={
                  Array [
                    "isEmail",
                  ]
                }
                value=""
                variant="standard"
              />,
              <TextValidator
                className="textArea"
                errorMessages="error"
                label="Message"
                multiline={true}
                name="message"
                onChange={[Function]}
                rows={6}
                validatorListener={[Function]}
                validators={Array []}
                value=""
                variant="standard"
              />,
              <WithStyles(Button)
                className="sendButton"
                color="primary"
                type="submit"
                variant="contained"
              >
                Send
              </WithStyles(Button)>,
            ],
            "className": "contactFormWrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "errorMessages": "error",
                "label": "Name",
                "name": "name",
                "onChange": [Function],
                "required": true,
                "validatorListener": [Function],
                "validators": Array [],
                "value": "",
                "variant": "standard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "errorMessages": Array [
                  "Email is not valid",
                ],
                "label": "Email",
                "name": "email",
                "onChange": [Function],
                "required": true,
                "validatorListener": [Function],
                "validators": Array [
                  "isEmail",
                ],
                "value": "",
                "variant": "standard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "textArea",
                "errorMessages": "error",
                "label": "Message",
                "multiline": true,
                "name": "message",
                "onChange": [Function],
                "rows": 6,
                "validatorListener": [Function],
                "validators": Array [],
                "value": "",
                "variant": "standard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Send",
                "className": "sendButton",
                "color": "primary",
                "type": "submit",
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Send",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Tile
          center={true}
        >
          <ValidatorForm
            debounceTime={0}
            onError={[Function]}
            onSubmit={[Function]}
          >
            <div
              className="contactFormWrapper"
            >
              <TextValidator
                errorMessages="error"
                label="Name"
                name="name"
                onChange={[Function]}
                required={true}
                validatorListener={[Function]}
                validators={Array []}
                value=""
                variant="standard"
              />
              <TextValidator
                errorMessages={
                  Array [
                    "Email is not valid",
                  ]
                }
                label="Email"
                name="email"
                onChange={[Function]}
                required={true}
                validatorListener={[Function]}
                validators={
                  Array [
                    "isEmail",
                  ]
                }
                value=""
                variant="standard"
              />
              <TextValidator
                className="textArea"
                errorMessages="error"
                label="Message"
                multiline={true}
                name="message"
                onChange={[Function]}
                rows={6}
                validatorListener={[Function]}
                validators={Array []}
                value=""
                variant="standard"
              />
              <WithStyles(Button)
                className="sendButton"
                color="primary"
                type="submit"
                variant="contained"
              >
                Send
              </WithStyles(Button)>
            </div>
          </ValidatorForm>
        </Tile>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "center": true,
          "children": <ValidatorForm
            debounceTime={0}
            onError={[Function]}
            onSubmit={[Function]}
          >
            <div
              className="contactFormWrapper"
            >
              <TextValidator
                errorMessages="error"
                label="Name"
                name="name"
                onChange={[Function]}
                required={true}
                validatorListener={[Function]}
                validators={Array []}
                value=""
                variant="standard"
              />
              <TextValidator
                errorMessages={
                  Array [
                    "Email is not valid",
                  ]
                }
                label="Email"
                name="email"
                onChange={[Function]}
                required={true}
                validatorListener={[Function]}
                validators={
                  Array [
                    "isEmail",
                  ]
                }
                value=""
                variant="standard"
              />
              <TextValidator
                className="textArea"
                errorMessages="error"
                label="Message"
                multiline={true}
                name="message"
                onChange={[Function]}
                rows={6}
                validatorListener={[Function]}
                validators={Array []}
                value=""
                variant="standard"
              />
              <WithStyles(Button)
                className="sendButton"
                color="primary"
                type="submit"
                variant="contained"
              >
                Send
              </WithStyles(Button)>
            </div>
          </ValidatorForm>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="contactFormWrapper"
            >
              <TextValidator
                errorMessages="error"
                label="Name"
                name="name"
                onChange={[Function]}
                required={true}
                validatorListener={[Function]}
                validators={Array []}
                value=""
                variant="standard"
              />
              <TextValidator
                errorMessages={
                  Array [
                    "Email is not valid",
                  ]
                }
                label="Email"
                name="email"
                onChange={[Function]}
                required={true}
                validatorListener={[Function]}
                validators={
                  Array [
                    "isEmail",
                  ]
                }
                value=""
                variant="standard"
              />
              <TextValidator
                className="textArea"
                errorMessages="error"
                label="Message"
                multiline={true}
                name="message"
                onChange={[Function]}
                rows={6}
                validatorListener={[Function]}
                validators={Array []}
                value=""
                variant="standard"
              />
              <WithStyles(Button)
                className="sendButton"
                color="primary"
                type="submit"
                variant="contained"
              >
                Send
              </WithStyles(Button)>
            </div>,
            "debounceTime": 0,
            "onError": [Function],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TextValidator
                  errorMessages="error"
                  label="Name"
                  name="name"
                  onChange={[Function]}
                  required={true}
                  validatorListener={[Function]}
                  validators={Array []}
                  value=""
                  variant="standard"
                />,
                <TextValidator
                  errorMessages={
                    Array [
                      "Email is not valid",
                    ]
                  }
                  label="Email"
                  name="email"
                  onChange={[Function]}
                  required={true}
                  validatorListener={[Function]}
                  validators={
                    Array [
                      "isEmail",
                    ]
                  }
                  value=""
                  variant="standard"
                />,
                <TextValidator
                  className="textArea"
                  errorMessages="error"
                  label="Message"
                  multiline={true}
                  name="message"
                  onChange={[Function]}
                  rows={6}
                  validatorListener={[Function]}
                  validators={Array []}
                  value=""
                  variant="standard"
                />,
                <WithStyles(Button)
                  className="sendButton"
                  color="primary"
                  type="submit"
                  variant="contained"
                >
                  Send
                </WithStyles(Button)>,
              ],
              "className": "contactFormWrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "errorMessages": "error",
                  "label": "Name",
                  "name": "name",
                  "onChange": [Function],
                  "required": true,
                  "validatorListener": [Function],
                  "validators": Array [],
                  "value": "",
                  "variant": "standard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "errorMessages": Array [
                    "Email is not valid",
                  ],
                  "label": "Email",
                  "name": "email",
                  "onChange": [Function],
                  "required": true,
                  "validatorListener": [Function],
                  "validators": Array [
                    "isEmail",
                  ],
                  "value": "",
                  "variant": "standard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "textArea",
                  "errorMessages": "error",
                  "label": "Message",
                  "multiline": true,
                  "name": "message",
                  "onChange": [Function],
                  "rows": 6,
                  "validatorListener": [Function],
                  "validators": Array [],
                  "value": "",
                  "variant": "standard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Send",
                  "className": "sendButton",
                  "color": "primary",
                  "type": "submit",
                  "variant": "contained",
                },
                "ref": null,
                "rendered": "Send",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
